### PR TITLE
feat: tutorial and comparison of ways to connect Casdoor

### PR DESCRIPTION
**Pull request by leo220yuyaodog**
close: #369
Write more details to help user select one way to connect Casdoor.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@Selflocking

File: docs/how-to-connect/overview.md [hunk 1]
```diff
@@ -11,16 +11,110 @@ As Service Provider (SP), Casdoor supports two authentication protocols:
 - `OAuth 2.0 (OIDC)`
 - `SAML`
 
-As Identity Provider (IdP), Casdoor supports one authentication protocol:
+As Identity Provider (IdP), Casdoor supports 4 authentication protocol:
```
Comment (yoyo314) [inline docs/how-to-connect/overview.md:14]: protocols

File: docs/how-to-connect/overview.md [hunk 2]
```diff
@@ -11,16 +11,110 @@ As Service Provider (SP), Casdoor supports two authentication protocols:
...
 - `CAS 1.0, 2.0, 3.0`
 
+## OAuth 2.0 (OIDC)
+
+<details>
+
+<summary>What is OAuth 2.0?</summary>
+
+[OAuth 2](https://oauth.net/2/) is an authorization framework that enables applications — such as Facebook, GitHub, and Casdoor — to obtain
+limited access to user accounts on an HTTP service. It works by delegating user authentication to the service that hosts
```
Comment (yoyo314) [inline docs/how-to-connect/overview.md:28]: a HTTP service

File: docs/how-to-connect/overview.md [hunk 3]
```diff
@@ -11,16 +11,110 @@ As Service Provider (SP), Casdoor supports two authentication protocols:
...
+
+<summary>What is SAML?</summary>
+
+Security Assertion Markup Language (SAML) is an open standard that allows identity providers (IdP) to pass authorization credentials to service providers (SP). What that jargon means is that you can use one set of credentials to log into many different websites. It’s much simpler to manage one login per user than it is to manage separate logins to email, customer relationship management (CRM) software, Active Directory, etc.
+
+SAML transactions use Extensible Markup Language (XML) for standardized communications between the identity provider and service providers. SAML is the link between the authentication of a user’s identity and the authorization to use a service.
+
+</details>
+
+Casdoor can be used as **SAML IdP**. Up to now the Casdoor have supported the main feature of **SAML 2.0**. More details see **[SAML](/docs/how-to-connect/saml)**.
```
Comment (yoyo314) [inline docs/how-to-connect/overview.md:87]: has supported the main features

File: docs/how-to-connect/overview.md [hunk 4]
```diff
@@ -11,16 +11,110 @@ As Service Provider (SP), Casdoor supports two authentication protocols:
...
+
+**Example:**
+
+[Casdoor as a SAML IdP in keycloak](/docs/how-to-connect/saml#casdoor-as-a-saml-idp-in-keycloak)
+
+**Suggestion:**
+
+1. The protocol is **powerful** and covers many scenarios, which can be said to be one of the most comprehensive SSO protocols.
+2. The protocol is **too large**, and there are many optional, so it is difficult to cover all application scenarios 100% in the actual implementation.
```
Comment (yoyo314) [inline docs/how-to-connect/overview.md:96]: optional is an adj, maybe there something missing after the word or "options"?

File: docs/how-to-connect/overview.md [hunk 5]
```diff
@@ -11,16 +11,110 @@ As Service Provider (SP), Casdoor supports two authentication protocols:
...
+
+**Suggestion:**
+
+1. The protocol itself is relatively lightweight and easy to implement, but it can solve a single scenario.
+2. The mutual trust between the CAS Client and the CAS Server is established through interface invocation without any encryption or signature mechanism to ensure further security.
+3. CAS protocol has no advantage over other protocols.
+
+## Integrations table
+
+Some applications already have examples that connect to Casdoor. You can follow the documentation to quickly connect Casdoor.
```
Comment (yoyo314) [inline docs/how-to-connect/overview.md:119]: quickly connect to Casdoor

**Review by yoyo314 — CHANGES_REQUESTED** [review]

**Pull request merged**